### ISSUE TITLE
[MU4 Task] Add functionality "Replace image"

### ISSUE THREAD

**Issue by MortenHe** [musescore/MuseScore#15946]
First of all. Great software, great team, great support. Thank you very much!

Here is my enheacement suggestion:

I sometimes add an image to a score like a logo that I designed in Photoshop.
When I want to replace the image (e.g. adjustig the graphic but keeping its dimensions), the current workflow is to delete the image an insert it again.

I'd like to have the option. "Right click on image -> replace image". So the image would stay in its place and only its content would be updated.

Would do you think?

Many greetings from Freiburg / Germany
Martin

**Comment by Salavin** [comment]
Would adding a button to the Properties tab be sufficient, like so? 
<img width="302" alt="Untitled" src="https://user-images.githubusercontent.com/26843753/213525089-582d9386-396a-477f-bb5d-d8e7409269e6.png">

**Comment by Salavin** [comment]
Sure!
<img width="302" alt="image" src="https://user-images.githubusercontent.com/26843753/213529394-b0501251-b01b-460e-b5e1-4c1518eaca9c.png">

**Comment by MortenHe** [comment]
The button in the properties tab looks great.

Maybe the "right click on image -> replcae image" could be added later.

But I'm already really glad with the button solution.

**Comment by iwoithe** [comment]
> Maybe the "right click on image -> replcae image" could be added later.

@bkunda What do you think of this?

**Comment by bkunda** [comment]
> > Maybe the "right click on image -> replcae image" could be added later.
> 
> @bkunda What do you think of this?

I think adding it to the context menu is a reasonable suggestion; it adds another (intuitive) option to discover this feature, and might also have some additional benefits for accessibility too.

I'd probably suggest positioning it in the top panel, together with other options for removing the selected image: 

<img width="286" alt="Mac" src="https://user-images.githubusercontent.com/86290556/214304671-b76f5499-5967-4ec5-9df2-de77b32ec02f.png">

But this doesn't have to be done right away if you want to keep things simple for this PR 🙂

**Comment by cbjeukendrup** [comment]
@iwoithe One remark; quoting the original issue description:

> e.g. adjusting the graphic but keeping its dimensions

> So the image would stay in its place and only its content would be updated.

From the code in your PR, it is not immediately clear to me that this will happen indeed. Would be great if you could make it work like that 🙂

Also, please make sure that it works correctly with undo / redo.

**Comment by iwoithe** [comment]
> But this doesn't have to be done right away if you want to keep things simple for this PR :slightly_smiling_face:

I'll stick to what is currently in the PR. Once that is merged, I'll open a new PR with the context menu item. Won't be difficult to do.

**Comment by iwoithe** [comment]
> From the code in your PR, it is not immediately clear to me that this will happen indeed. Would be great if you could make it work like that :slightly_smiling_face:

Thanks for pointing that out. Kind of overlooked that :roll_eyes:

Though I wonder if having the option for both would be useful. Something like this?

![REPLACE IMAGE BUTTON](https://user-images.githubusercontent.com/73422868/215373771-3e8a57b2-ca25-4c57-8101-adab7666ef95.png)

**Comment by cbjeukendrup** [comment]
@bkunda What do you think about the suggestion above?

**Comment by bkunda** [comment]
> @bkunda What do you think about the suggestion above?

Honestly, I can't make up my mind about this one. 
I admit I was rather fond of the simplicity of just clicking **Replace image** and seeing the OS file manager dialog. The duplication of the "Replace image" label is potentially a little jarring, but at the same time it's also not really ambiguous. 

I'd like a some more designer opinions on this @Tantacrul @jessjwilliamson.

**Comment by Tantacrul** [comment]
I quite like the idea of having it in Properties (with the additional options @iwoithe showed). I also think it should be available in the context menu too.

To be a design / product pedant: unique options should never be in the context menu. The context menu should only be a convenient way of accessing options already found elsewhere.

**Comment by Tantacrul** [comment]
For me though, I'd opt for a little floating widget above an image when you click on it that says 'Replace image'.

You see this all over the place. SquareSpace do it really well when you are editing images for your website. Just click on them and you see results like this above:

<img width="271" alt="image" src="https://user-images.githubusercontent.com/21022311/225623886-a62a3e64-5d4a-4679-b9d1-f949e2c8df75.png">

**Comment by Tantacrul** [comment]
However, we can always do the floating thing later. Properties + context menu is more than enough for now.

**Comment by bkunda** [comment]
This is good enough for me 🙂.

@iwoithe let's proceed with your suggestion of the options in Properties, and if you want to also add the context menu option then go ahead. 

I promise I won't be so tardy with testing as I was with replying to this conversation!

**Comment by Tantacrul** [comment]
I would say you should add the replace image option in the context menu (obviously only when RC on an image!). It'd be swell! Thanks all!
